smart_contracts/contract/src/ext_ffi.rs

```diff
@@ -767,6 +767,22 @@ extern "C" {
         value_ptr: *const u8,
         value_size: usize,
     ) -> i32;
+    /// The bytes in the span of wasm memory from `key_ptr` to `key_ptr + key_size` must correspond
+    /// to a valid global state dictionary key, otherwise the function will fail.
+    /// If the key is de-serialized successfully, then the result of the read is serialized
+    /// and buffered in the runtime. This result can be obtained via the [`casper_read_host_buffer`]
+    /// function. Returns standard error code.
+    ///
+    /// # Arguments
+    ///
+    /// * `key_ptr` - pointer (offset in wasm linear memory) to serialized form of the key to read
+    /// * `key_size` - size of the serialized key (in bytes)
+    /// * `output_size` - pointer to a value where host will write size of bytes read from given key
+    pub fn casper_dictionary_read(
```

Comment:
put next to dictionary_get above

since both the dictionary_item_key expected by dictionary_get and the actual key expected by this ffi method both have the same arg names (key_ptr, key_size) you should make very clear in the comments on 778 and 779 that this expects a Key::Dictionary

Reply:
Moved to `dictionary_get` and made the comments more clear https://github.com/casper-network/casper-node/pull/3161/commits/862d2b060f8391957da23d76025c2b8a6fd92c5f